ml-agents/mlagents/trainers/agent_processor.py

```diff
@@ -156,3 +156,49 @@ def add_experiences(
         self.policy.save_previous_action(
             curr_info.agents, take_action_outputs["action"]
         )
+
+    def publish_trajectory_queue(self, trajectory_queue: Queue) -> None:
+        """
+        Adds a trajectory queue to the list of queues to publish to when this AgentProcessor
+        assembles a Trajectory
+        :param trajectory_queue: Trajectory queue to publish to.
+        """
+        self.trajectory_queues.append(trajectory_queue)
+
+
+class AgentManagerQueue:
+    """
+    Queue used by the AgentManager. Note that we make our own class here because in most implementations
+    deque is sufficient and faster. However, if we want to switch to multiprocessing, we'll need to change
+    out this implementation.
+    """
+
+    def __init__(self, behavior_id: str = None):
+        """
+        Initializes an AgentManagerQueue. Note that we can give it a behavior_id so that it can be identified
+        separately from an AgentManager.
+        """
+        self.queue: Deque[Union[Trajectory, Policy]] = deque()
+        self.behavior_id = behavior_id
+
+    def empty(self) -> bool:
+        return len(self.queue) == 0
+
+    def get_nowait(self) -> Union[Trajectory, Policy]:
+        return self.queue.popleft()
+
+    def put(self, item: Union[Trajectory, Policy]) -> None:
+        self.queue.append(item)
+
+
+class AgentManager(AgentProcessor):
+    """
+    An AgentManager is an AgentProcessor that also holds a single trajectory and policy queue.
+    Note: this leaves room for adding AgentProcessors that publish multiple trajectory queues.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```

Comment:
I'd prefer to be explicit about the arguments here, but if you think it's too tedious, ignore me.

Reply:
Explicit it is